### PR TITLE
minor: Add input file for Java14 instanceof with pattern matching (#7290)

### PR DESCRIPTION
**Pull request by nmancus1**
Issue #7290: Add input file for Java14 instanceof with pattern matching

Example code taken from:
https://jaxenter.com/pattern-matching-for-instanceof-in-java-14-169830.html
https://www.baeldung.com/java-pattern-matching-instanceof
https://openjdk.java.net/jeps/305

The last three code snippets were troublesome in testing grammar, so I included them even though they really aren't pattern matching examples.

### CODE REVIEW

**Comment by nmancus1** [comment]
@esilkensen ping

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java14/InputJava14InstanceofWithPatternMatching.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package com.puppycrawl.tools.checkstyle.grammar.java14;
```
Comment (pbludov) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java14/InputJava14InstanceofWithPatternMatching.java:1]: There should be `non-compiled` marker. See
https://github.com/checkstyle/checkstyle/blob/master/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java9/InputJava9TryWithResources.java#L1 for reference

Comment (nmancus1) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java14/InputJava14InstanceofWithPatternMatching.java:2]: Fixed.

**Comment by esilkensen** [comment]
@romani, similar question here - should I go ahead and merge a PR like this one if @pbludov has already approved, so to not add another thing to your queue + slow down @nmancus1?

**Comment by romani** [comment]
For Inputs, and other changes that are not affecting checkstyle execution, you can quickly merge PRs.
For other functionality is better to wait for me and @rnveach , to make sure that all testing is done. 
FYI, we already missed big regression in 8.33 release after grammar change - https://github.com/checkstyle/checkstyle/issues/8299 , it is my fault that I did not spend that  much time on that PR to verify it with attention.

File: config/ant-phase-verify.xml [hunk 2]
```diff
@@ -148,6 +148,9 @@
          <exclude name="**/InputMainFrameModelIncorrectClass.java"/>
          <exclude name="**/InputBeforeExecutionExclusionFileFilterIncorrectClass.java"/>
          <exclude name="**/InputJavaParser.java"/>
+
+
+         <exclude name="**/InputJava14InstanceofWithPatternMatching.java"/>
```
Comment (romani) [inline config/ant-phase-verify.xml:153]: can we avoid empty lines ?
please add comment `until http://....../7290`

**Pull request merged**